concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -432,19 +448,30 @@ exportConsensusV1Blocks outDir chunkSize genIndex startHeight blockIndex lastWri
                             logEvent External LLError "Cannot read last block of the database."
                             return (True, Empty)
                         Just sb -> do
-                            let getBlockAt' :: BlockHeight -> m (Maybe BS.ByteString)
-                                getBlockAt' height =
+                            let getBlockAt :: BlockHeight -> m (Maybe (BS.ByteString, BlockHash))
+                                getBlockAt height =
                                     KonsensusV1.resizeOnResized
                                         (KonsensusV1.lookupBlockByHeight height)
                                         >>= \case
                                             Nothing -> return Nothing
                                             Just b | KonsensusV1.NormalBlock signedBlock <- KonsensusV1.stbBlock b -> do
                                                 let serializedBlock = runPut $ KonsensusV1.putSignedBlock signedBlock
-                                                return $ Just serializedBlock
+                                                return $ Just (serializedBlock, getHash signedBlock)
                                             _ -> return Nothing -- Do not export genesis blocks.
-                                getBlockAt = GetBlockAtV1 getBlockAt'
+                                getFinalizationAt :: BlockHash -> m (Maybe BS.ByteString)
```

Comment:
It'd be useful to comment here what the semantics/intent here is with the Maybe and BlockHash.